gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupRequestHandler.java

```diff
@@ -189,6 +193,31 @@ private List<BackupStatus> aggregateBackupList(
         .toList();
   }
 
+  // When a backup status returns more than one status for a partition, this method helps to
+  // choose the best one from the available backups of a partition.
+  private BackupListResponse.BackupStatus mergeDuplicatePartitionBackupStatus(
+      final BackupListResponse.BackupStatus x, final BackupListResponse.BackupStatus y) {
+    if (x.partitionId() != y.partitionId()) {
+      throw new IllegalArgumentException(
+          "Expected to merge backup status from same partitions, but provided backups of different partitions. Provided backups : %s, %s"
+              .formatted(x, y));
+    }
+
+    final List<BackupStatusCode> comparingOrder =
+        List.of(
+            BackupStatusCode.SBE_UNKNOWN,
+            BackupStatusCode.DOES_NOT_EXIST,
+            BackupStatusCode.FAILED,
+            BackupStatusCode.IN_PROGRESS,
+            BackupStatusCode.COMPLETED);
```

Comment:
:wrench: Maybe we should have this statically defined in `BackupListResponse.BackupStatus` or as a helper method? So that this merge logic could just be

```java
if (x.hasBetterStatusThan(y)) {
  return x;
} else {
  return y;
}
```

Reply:
`BackupStatus` is generated code from SBE.

Reply:
Hm `BackupListResponse.BackupStatus` isn't, we define it here: https://github.com/camunda/zeebe/blob/81ccf953228d0dc77ae7937b174afd02f28e5803/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupListResponse.java#L119-L125

Reply:
Oops. Sorry. I thought you meant BackupStatusCode.